content/posts/flp-inside-out.md

```diff
@@ -0,0 +1,78 @@
+
+This might not be obvious at the first sight. Lets start by look at the set of all inital configurations for 3 processes. An initial configuration is fully specified by values of 3 `x` registers each storing either `0` or `1`. There are 2^3 initial configurations that we can represent as 3-digit binary number. A chain of initial configurations in which one differs by a single bit is simply a enumarion of [Gray coding](TODO: add ref) for 3-digit binary number. In our case, `000` -> `001` -> `011` -> `010` -> `110` -> `101` -> `100`.
+
+If the chain contains both 1-valent and 0-valent intial configuration it's trivial to see that we can travers the sequence from one to the other and must arrive at an edge that connects 1-valent and 0-valent configurations that differ by a single bit i.e. by a single `x` register.
+
+//TODO: include a tesseract of 
+
+#### Internal state of the processes are not important
+
+But what about internal state of the process? Why isn't it mentioned in anywhere in the lemma? Does this imply that each process must start with the same internal state? Can we "break" the proof by coming up with an alogrithm that starts off with processes having different internal state? 
+
+The answer is ... we don't care if all processes have the same internal state or not. If an alogrithm is deterministic it has to define the internal state of each process before any of the values for `x` registers get defined. Futhermore, the state can depend only on some knowledge available before the algorithm starts e.g. process identifier. What this means is that even if each process has different internal state, these (different) states will be same for each possible permutation of the `x` register values if the algorithm is deterministic. 
+
+
+### Lemma 3
+
+#### C0 and C1 that are one step apart
+
+> (...) D contains both 0-valent and 1-valent configurations.
+> Call two configurations neighbors if one results from the other in a single step. By an easy induction, there exist neighbors Co, Ci E, such that Di = e(Ci) is i-valent, i = 0, 1.
+
+Why is it true that there have to be C0 and C1 that are only one step aparts? Well if C is the configuration we start from than all reachable configurations through schedules without `e` form a DAG. Nodes are configurations of the system and edges connect configurations if there is an event that can lead from configuraiton to the other. D as defined in the paper can be "visialized" as a layer of nodes atop the DAG (reachable via `e` event). We already know that D contains both 0-valent and 1-valent configuration (let's call them D0' and D1'). Let's consider a path between C0' and C1' nodes that connect to D0' and D1' via `e`. There must be an edge `e'` on that path that connects nodes with neighroubs in D that have different valence.
```

Comment:
Nitpicking: then instead of than in "...from than all reachable configurations through..."

Reply:
Nitpicking: visialized (typo)

Reply:
Nitpicking: configuraiton (typo)

Reply:
Nitpicking: neighroubs (typo)